Enable the license checker over distribution/* and plugins/*

clintongormley opened the pull request:
Moved the license checker config into the parent pom, and overrode
the license dir/target-to-check in distributions/pom.

Disabled the license checker explicitly for projects which run integration
tests but have no licenses dir:
- core
- distribution
- qa

Closes #12752

clintongormley commented:
mvn verify works for me

dadoonet commented on core/pom.xml:
```diff
@@ -320,6 +320,18 @@
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-antrun-plugin</artifactId>
+                <version>1.8</version>
```
Don't set the version here. It should be managed by the Parent pom.

clintongormley committed:
Enable the license checker over distribution/* and plugins/*

Moved the license checker config into the parent pom, and overrede
the license dir/target-to-check in distributions/pom.

Disabled the license checker explicitly for projects which run integration
tests but have no licenses dir:

* core
* distribution
* qa
* plugins/delete-by-query
* plugins/mapper-size
* plugins/site-example

Closes #12752
Closes #12754

clintongormley commented:
thanks @dadoonet 

updated to only specify the version in the parent pom

dadoonet commented:
It looks good to me.
I really like we enable this only when it makes sense instead of disabling when it does not.

However, I'm not a big fan of disabling this in plugins when no 3rd party lib is used.
I think we should have this `licenses` dir in every plugin but with an "empty" file like `no_lib.txt` to fix the git issue with empty dirs.

Because if at some point we need to add a lib to one of the existing plugin, we might forget to reenable this license check.

WDYT?

clintongormley committed:
Enable the license checker for plugins without third party deps too

clintongormley commented:
> Because if at some point we need to add a lib to one of the existing plugin, we might forget to reenable this license check.

Makes sense. I've pushed a new commit, and will rerun mvn verify.

dadoonet commented:
LGTM

clintongormley committed:
Removed perl 5.8 support from the license checker

No point in keeping 5.8 support because it relies on modules
that are shipped with 5.9.
